sentry/src/main/java/io/sentry/SynchronizedCollection.java

```diff
@@ -50,7 +52,7 @@ class SynchronizedCollection<E> implements Collection<E>, Serializable {
   /** The collection to decorate */
   private final Collection<E> collection;
   /** The object to lock on, needed for List/SortedSet views */
-  final Object lock;
+  final AutoClosableReentrantLock lock;
```

Comment:
could this be `private` as well?

Reply:
No because it's used in `SynchronizedQueue`. The same is basically true for any non final class since you have to use the right lock instead of just relying on the synchronized keyword on methods.